crates/rmf_site_editor/src/site/save.rs

```diff
@@ -1425,12 +1429,17 @@ fn generate_scenarios(
                             Scenario {
                                 instances: scenario_instance_modifiers
                                     .iter()
-                                    .filter_map(|child_entity| {
-                                        instance_modifiers.get(*child_entity).ok()
-                                    })
-                                    .filter_map(|(pose, visibility, affiliation)| {
-                                        Some((
-                                            affiliation.0.and_then(|e| instances.get(e).ok())?.0,
+                                    .map(|e| {
+                                        let (pose, visibility, affiliation) = instance_modifiers
+                                            .get(*e)
+                                            .map_err(|_| SiteGenerationError::BrokenModifier(*e))?;
+                                        let a = affiliation
+                                            .0
+                                            .ok_or(SiteGenerationError::EmptyAffiliation(*e))?;
+                                        Ok((
+                                            instances.get(a).map(|id| id.0).map_err(|_| {
+                                                SiteGenerationError::BrokenAffiliation(a)
+                                            })?,
```

Comment:
```suggestion
                                        let a = affiliation
                                            .0
                                            .ok_or(SiteGenerationError::EmptyAffiliation(*e))?;
                                        let id = instances.get(a).map(|id| id.0).map_err(|_| {
                                            SiteGenerationError::BrokenAffilation(a)
                                        })?;
                                        Ok((
                                            id,
```

Big `Ok` statements are a bit of a pain to read